src/app/services/auth/auth.service.ts

```diff
@@ -39,10 +40,13 @@ export class AuthService {
   }
 
   login({ login, password }: LoginData) {
+    this.loadingService.startLoading();
+
     return this.http.post<LoginResponse>(`${APIUrl}/auth/login`, { login, password }).pipe(
       catchError(({ error }) => throwError(error)),
       tap(({ token }) => this.setUserDataToLocalStorage(login, token)),
       tap(() => this.emitAuthData()),
+      finalize(() => this.loadingService.stopLoading()),
```

Comment:
take a look at https://github.com/boale/interceptors/tree/master/src/app/interceptors
there you can find a sample of how to implement the same hide/show loader features through an interceptor.

Reply:
Thanks, nice approach